src/network/websockets/server/index.ts

```diff
@@ -53,6 +53,22 @@ export function runWebsocketServer<AuthDetails>(params: Params<AuthDetails>) {
       socket.send(JSON.stringify(obj));
     };
 
+    let preAuthMessageQueue: string[] = [];
+
+    const processQueue = () => {
+      preAuthMessageQueue.forEach(handleMsg);
+      preAuthMessageQueue = [];
+    };
+
+    const addToQueue = (msg: any) => {
+      if (preAuthMessageQueue.length > 100) {
+        console.warn("Cache Size exceeded maximum allowed length");
```

Comment:
Should we print some kind of client identifier (ID, username, IP address) in this message as well so we can differentiate between an offending user versus a defect in a release of the app?

(I forget if that's in the `terminate()` call. Any unexpected termination might be a better place to record that.)

Reply:
I think the server should not queue up any messages if we are not authenticated yet in order to avoid the attack vectors we discussed earlier today. I would be more comfortable if the queue happened on the oncilla client side.

Reply:
> I'm not going to block the PR over this, but I'm going to write down what I said on today's call:
> 
> My architectural recommendation would be to move the queue to the client side. This eliminates the attack vector where a malicious user could open a connection and send messages to fill up the queue, thus running the server out of memory. I do see that we've limited the max number of messages to 100, but an attacker could figure out that number based on when the termination happens, so it doesn't necessarily mitigate an attack (the attacker simply opens multiple connections to fill up the server's memory). Conversely, that server-side queue limit could cause UX issues for a legitimate user who is experiencing latency issues during the auth process. If the queue were moved client-side it would not only help with auth, but also if the connection goes offline for any other reason. (Related to [stagger-offline-ui-changes](https://trello.com/c/4aWviTri/3-stagger-offline-ui-changes))

I've created an issue to track: https://github.com/citrusbyte/oncilla/issues/42